a3fe/configuration/engine_config.py

```diff
@@ -5,298 +5,243 @@
-                lines[i] = "ncycles = " + str(n_cycles) + "\n"
-                break
-        #Now write the new file
-        with open(_os.path.join(self.input_dir, "somd.cfg"), "w+") as ofile:
-            for line in lines:
-                ofile.write(line)
-
-    def _validate_runtime_and_update_config(self) -> None:
-        """
-        Validate the runtime and update the simulation configuration.
+    ### Minimisation ###
+    minimise: bool = _Field(True, description="Perform energy minimisation")
 
-        Need to make sure that runtime is a multiple of the time per cycle
-        otherwise actual time could be quite different from requested runtime
+    ### Restraints ###
+    use_boresch_restraints: bool = _Field(False, description="Use Boresch restraints mode")
+    turn_on_receptor_ligand_restraints: bool = _Field(False, description="Turn on receptor-ligand restraints mode")
 
-        Raises
-        ------
-        ValueError
-            If runtime is not a multiple of the time per cycle.
-        """
-        time_per_cycle = self.timestep * self.nmoves / 1_000_000  # Convert fs to ns
-        # Convert both to float for division
-        remainder = float(self.runtime) / float(time_per_cycle)
-        if not _isclose(remainder - round(remainder), 0, abs_tol=1e-5):
+    ### Alchemistry - restraints added by a3fe ###
+    perturbed_residue_number: int = _Field(1,alias="perturbed residue number",ge=1, description="Residue number to perturb. Must be >= 1")
+    energy_frequency: int = _Field(200,alias="energy frequency",description="Frequency of energy output")
+    ligand_charge: int = _Field(0, description="Net charge of the ligand. If non-zero, must use PME for electrostatics.")
+    charge_difference: int = _Field(0, description="Charge difference between the ligand and the system. If non-zero, must use co-alchemical ion approach.")
+    
+    ### Lambda ###
+    lambda_array: _List[float] = _Field(
+        default_factory=_get_default_lambda_array,
+        description="Lambda array for alchemical perturbation, varies from 0.0 to 1.0 across stages"
+    )
+    lambda_val: _Optional[float] = _Field(None, description="Lambda value for current stage")
+
+    ### Alchemical files ###
+    morphfile: _Optional[str] = _Field(None, description="Path to morph file containing alchemical transformation")
+    topfile: _Optional[str] = _Field(None, description="Path to topology file for the system")
+    crdfile: _Optional[str] = _Field(None, description="Path to coordinate file for the system")
+
+    boresch_restraints_dictionary: _Optional[str] = _Field(
+        None, 
+        #alias="boresch restraints dictionary",
+        description="Optional string to hold boresch restraints dictionary content"
+    )    
+    ### Extra options ###
+    extra_options: _Dict[str, str] = _Field(default_factory=dict, description="Extra options to pass to the SOMD engine")    
+
+    @_field_validator('runtime')
```

Comment:
Please use a model validator - this is more suited to comparing multiple fields. Thanks. https://docs.pydantic.dev/latest/concepts/validators/#model-validators